{
  "cells": [
    {
      "cell_type": "code",
      "source": [
        "# @title Install AlphaGenome\n",
        "\n",
        "# @markdown Run this cell to install AlphaGenome.\n",
        "from IPython.display import clear_output\n",
        "! pip install alphagenome\n",
        "clear_output()"
      ],
      "metadata": {
        "id": "8ZTL88lbDUws"
      },
      "execution_count": 1,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# @title Setup and imports.\n",
        "\n",
        "from io import StringIO\n",
        "from alphagenome import colab_utils\n",
        "from alphagenome.data import genome\n",
        "from alphagenome.models import dna_client, variant_scorers\n",
        "from google.colab import data_table, files\n",
        "import pandas as pd\n",
        "from tqdm import tqdm\n",
        "\n",
        "data_table.enable_dataframe_formatter()\n",
        "\n",
        "# Load the model.\n",
        "dna_model = dna_client.create(colab_utils.get_api_key())"
      ],
      "metadata": {
        "id": "wADgbzZSDf1m"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# @title Score batch of variants with aggregation\n",
        "\n",
        "# Load VCF file containing variants.\n",
        "vcf_file = 'brca1.csv'  # @param\n",
        "\n",
        "\n",
        "vcf = pd.read_csv(vcf_file, sep=',') # Changed separator to comma\n",
        "vcf = vcf.rename(columns={\n",
        "    'chr': 'CHROM',\n",
        "    'pos': 'POS',\n",
        "    'ref': 'REF',\n",
        "    'alt': 'ALT'\n",
        "}) # Rename columns to match expected names\n",
        "# Create variant_id column\n",
        "vcf['variant_id'] = vcf['CHROM'].astype(str) + ':' + \\\n",
        "                    vcf['POS'].astype(str) + ':' + \\\n",
        "                    vcf['REF'].astype(str) + '>' + \\\n",
        "                    vcf['ALT'].astype(str)\n",
        "\n",
        "required_columns = ['variant_id', 'CHROM', 'POS', 'REF', 'ALT']\n",
        "for column in required_columns:\n",
        "  if column not in vcf.columns:\n",
        "    raise ValueError(f'VCF file is missing required column: {column}.')\n",
        "\n",
        "organism = 'human'  # @param [\"human\", \"mouse\"] {type:\"string\"}\n",
        "\n",
        "# @markdown Specify length of sequence around variants to predict:\n",
        "sequence_length = '1MB'  # @param [\"16KB\", \"100KB\", \"500KB\", \"1MB\"] { type:\"string\" }\n",
        "sequence_length = dna_client.SUPPORTED_SEQUENCE_LENGTHS[\n",
        "    f'SEQUENCE_LENGTH_{sequence_length}'\n",
        "]\n",
        "\n",
        "# @markdown Specify which scorers to use to score your variants:\n",
        "score_rna_seq = True  # @param { type: \"boolean\"}\n",
        "score_cage = True  # @param { type: \"boolean\" }\n",
        "score_procap = True  # @param { type: \"boolean\" }\n",
        "score_atac = True  # @param { type: \"boolean\" }\n",
        "score_dnase = True  # @param { type: \"boolean\" }\n",
        "score_chip_histone = True  # @param { type: \"boolean\" }\n",
        "score_chip_tf = True  # @param { type: \"boolean\" }\n",
        "score_polyadenylation = True  # @param { type: \"boolean\" }\n",
        "score_splice_sites = True  # @param { type: \"boolean\" }\n",
        "score_splice_site_usage = True  # @param { type: \"boolean\" }\n",
        "score_splice_junctions = True  # @param { type: \"boolean\" }\n",
        "\n",
        "# @markdown Other settings:\n",
        "download_predictions = True  # @param { type: \"boolean\" }\n",
        "\n",
        "# Parse organism specification.\n",
        "organism_map = {\n",
        "    'human': dna_client.Organism.HOMO_SAPIENS,\n",
        "    'mouse': dna_client.Organism.MUS_MUSCULUS,\n",
        "}\n",
        "organism = organism_map[organism]\n",
        "\n",
        "# Parse scorer specification.\n",
        "scorer_selections = {\n",
        "    'rna_seq': score_rna_seq,\n",
        "    'cage': score_cage,\n",
        "    'procap': score_procap,\n",
        "    'atac': score_atac,\n",
        "    'dnase': score_dnase,\n",
        "    'chip_histone': score_chip_histone,\n",
        "    'chip_tf': score_chip_tf,\n",
        "    'polyadenylation': score_polyadenylation,\n",
        "    'splice_sites': score_splice_sites,\n",
        "    'splice_site_usage': score_splice_site_usage,\n",
        "    'splice_junctions': score_splice_junctions,\n",
        "}\n",
        "\n",
        "all_scorers = variant_scorers.RECOMMENDED_VARIANT_SCORERS\n",
        "selected_scorers = [\n",
        "    all_scorers[key]\n",
        "    for key in all_scorers\n",
        "    if scorer_selections.get(key.lower(), False)\n",
        "]\n",
        "\n",
        "# Remove any scorers or output types that are not supported for the chosen organism.\n",
        "unsupported_scorers = [\n",
        "    scorer\n",
        "    for scorer in selected_scorers\n",
        "    if (\n",
        "        organism.value\n",
        "        not in variant_scorers.SUPPORTED_ORGANISMS[scorer.base_variant_scorer]\n",
        "    )\n",
        "    | (\n",
        "        (scorer.requested_output == dna_client.OutputType.PROCAP)\n",
        "        & (organism == dna_client.Organism.MUS_MUSCULUS)\n",
        "    )\n",
        "]\n",
        "if len(unsupported_scorers) > 0:\n",
        "  print(\n",
        "      f'Excluding {unsupported_scorers} scorers as they are not supported for'\n",
        "      f' {organism}.'\n",
        "  )\n",
        "  for unsupported_scorer in unsupported_scorers:\n",
        "    selected_scorers.remove(unsupported_scorer)\n",
        "\n",
        "\n",
        "# Score variants in the VCF file.\n",
        "results = []\n",
        "\n",
        "for i, vcf_row in tqdm(vcf.iterrows(), total=len(vcf)):\n",
        "  variant = genome.Variant(\n",
        "      chromosome=str(vcf_row.CHROM),\n",
        "      position=int(vcf_row.POS),\n",
        "      reference_bases=vcf_row.REF,\n",
        "      alternate_bases=vcf_row.ALT,\n",
        "      name=vcf_row.variant_id,\n",
        "  )\n",
        "  interval = variant.reference_interval.resize(sequence_length)\n",
        "\n",
        "  variant_scores = dna_model.score_variant(\n",
        "      interval=interval,\n",
        "      variant=variant,\n",
        "      variant_scorers=selected_scorers,\n",
        "      organism=organism,\n",
        "  )\n",
        "  results.append(variant_scores)\n",
        "\n",
        "df_scores = variant_scorers.tidy_scores(results)\n",
        "\n",
        "# Save original scores\n",
        "if download_predictions:\n",
        "  df_scores.to_csv('variant_scores.csv', index=False)\n",
        "  files.download('variant_scores.csv')\n",
        "\n",
        "# Aggregate scores by variant_id\n",
        "# Calculate max, min, and mean for both raw_score and quantile_score\n",
        "# Convert variant_id to string type to ensure hashability for groupby operation\n",
        "df_scores['variant_id'] = df_scores['variant_id'].astype(str)\n",
        "\n",
        "agg_max = df_scores.groupby('variant_id').agg({\n",
        "    'raw_score': 'max',\n",
        "    'quantile_score': 'max'\n",
        "}).reset_index()\n",
        "agg_max.columns = ['variant_id', 'raw_score_max', 'quantile_score_max']\n",
        "\n",
        "agg_min = df_scores.groupby('variant_id').agg({\n",
        "    'raw_score': 'min',\n",
        "    'quantile_score': 'min'\n",
        "}).reset_index()\n",
        "agg_min.columns = ['variant_id', 'raw_score_min', 'quantile_score_min']\n",
        "\n",
        "agg_mean = df_scores.groupby('variant_id').agg({\n",
        "    'raw_score': 'mean',\n",
        "    'quantile_score': 'mean'\n",
        "}).reset_index()\n",
        "agg_mean.columns = ['variant_id', 'raw_score_mean', 'quantile_score_mean']\n",
        "\n",
        "# Save aggregated scores\n",
        "if download_predictions:\n",
        "  agg_max.to_csv('variant_scores_max.csv', index=False)\n",
        "  files.download('variant_scores_max.csv')\n",
        "\n",
        "  agg_min.to_csv('variant_scores_min.csv', index=False)\n",
        "  files.download('variant_scores_min.csv')\n",
        "\n",
        "  agg_mean.to_csv('variant_scores_mean.csv', index=False)\n",
        "  files.download('variant_scores_mean.csv')\n",
        "\n",
        "print(\"\\nAggregated scores (max):\")\n",
        "display(agg_max)\n",
        "print(\"\\nAggregated scores (min):\")\n",
        "display(agg_min)\n",
        "print(\"\\nAggregated scores (mean):\")\n",
        "display(agg_mean)\n",
        "\n",
        "df_scores"
      ],
      "metadata": {
        "id": "-4K_lqt1DiqN",
        "outputId": "9c2f146b-bf62-465a-9324-736eedae1c33",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "100%|██████████| 1994/1994 [37:36<00:00,  1.13s/it]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "7FYdDiK4EZKr",
        "outputId": "016a03ae-da08-499e-f35b-d85cb8b3c124",
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 636
        }
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Warning: Total number of columns (24) exceeds max_columns (20). Falling back to pandas display.\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "              variant_id           scored_interval          gene_id  \\\n",
              "0      chr3:58394738:A>T  chr3:57870450-58919026:.             None   \n",
              "168    chr3:58394738:A>T  chr3:57870450-58919026:.             None   \n",
              "2109   chr3:58394738:A>T  chr3:57870450-58919026:.             None   \n",
              "2110   chr3:58394738:A>T  chr3:57870450-58919026:.             None   \n",
              "2111   chr3:58394738:A>T  chr3:57870450-58919026:.             None   \n",
              "...                  ...                       ...              ...   \n",
              "91363  chr16:1135446:G>T    chr16:611158-1659734:.  ENSG00000290756   \n",
              "91364  chr16:1135446:G>T    chr16:611158-1659734:.  ENSG00000292400   \n",
              "91365  chr16:1135446:G>T    chr16:611158-1659734:.  ENSG00000292423   \n",
              "91366  chr16:1135446:G>T    chr16:611158-1659734:.  ENSG00000292431   \n",
              "91367  chr16:1135446:G>T    chr16:611158-1659734:.  ENSG00000292432   \n",
              "\n",
              "             gene_name gene_type gene_strand junction_Start junction_End  \\\n",
              "0                 None      None        None           None         None   \n",
              "168               None      None        None           None         None   \n",
              "2109              None      None        None           None         None   \n",
              "2110              None      None        None           None         None   \n",
              "2111              None      None        None           None         None   \n",
              "...                ...       ...         ...            ...          ...   \n",
              "91363  ENSG00000290756    lncRNA           -           None         None   \n",
              "91364  ENSG00000292400    lncRNA           -           None         None   \n",
              "91365  ENSG00000292423    lncRNA           -           None         None   \n",
              "91366  ENSG00000292431    lncRNA           -           None         None   \n",
              "91367  ENSG00000292432    lncRNA           -           None         None   \n",
              "\n",
              "        output_type                                     variant_scorer  ...  \\\n",
              "0              ATAC  CenterMaskScorer(requested_output=ATAC, width=...  ...   \n",
              "168           DNASE  CenterMaskScorer(requested_output=DNASE, width...  ...   \n",
              "2109   CHIP_HISTONE  CenterMaskScorer(requested_output=CHIP_HISTONE...  ...   \n",
              "2110   CHIP_HISTONE  CenterMaskScorer(requested_output=CHIP_HISTONE...  ...   \n",
              "2111   CHIP_HISTONE  CenterMaskScorer(requested_output=CHIP_HISTONE...  ...   \n",
              "...             ...                                                ...  ...   \n",
              "91363       RNA_SEQ        GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
              "91364       RNA_SEQ        GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
              "91365       RNA_SEQ        GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
              "91366       RNA_SEQ        GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
              "91367       RNA_SEQ        GeneMaskLFCScorer(requested_output=RNA_SEQ)  ...   \n",
              "\n",
              "      biosample_type biosample_life_stage data_source endedness  \\\n",
              "0       primary_cell                adult      encode    paired   \n",
              "168     primary_cell                adult      encode    paired   \n",
              "2109    primary_cell                adult      encode    single   \n",
              "2110    primary_cell                adult      encode    single   \n",
              "2111    primary_cell                adult      encode    single   \n",
              "...              ...                  ...         ...       ...   \n",
              "91363   primary_cell                adult      encode    single   \n",
              "91364   primary_cell                adult      encode    single   \n",
              "91365   primary_cell                adult      encode    single   \n",
              "91366   primary_cell                adult      encode    single   \n",
              "91367   primary_cell                adult      encode    single   \n",
              "\n",
              "      genetically_modified transcription_factor histone_mark gtex_tissue  \\\n",
              "0                    False                  NaN          NaN         NaN   \n",
              "168                  False                  NaN          NaN         NaN   \n",
              "2109                 False                  NaN      H3K27ac         NaN   \n",
              "2110                 False                  NaN     H3K27me3         NaN   \n",
              "2111                 False                  NaN     H3K36me3         NaN   \n",
              "...                    ...                  ...          ...         ...   \n",
              "91363                False                  NaN          NaN               \n",
              "91364                False                  NaN          NaN               \n",
              "91365                False                  NaN          NaN               \n",
              "91366                False                  NaN          NaN               \n",
              "91367                False                  NaN          NaN               \n",
              "\n",
              "      raw_score quantile_score  \n",
              "0     -0.004951      -0.259272  \n",
              "168   -0.030459      -0.680793  \n",
              "2109  -0.000978      -0.301775  \n",
              "2110   0.000441       0.080577  \n",
              "2111   0.000000      -0.092027  \n",
              "...         ...            ...  \n",
              "91363 -0.000391      -0.402670  \n",
              "91364  0.000173       0.280661  \n",
              "91365 -0.001031      -0.641886  \n",
              "91366  0.000020       0.103452  \n",
              "91367  0.000900       0.606708  \n",
              "\n",
              "[556 rows x 24 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-1b643293-2470-4bc3-b01f-3d0ef6f4ecf2\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>variant_id</th>\n",
              "      <th>scored_interval</th>\n",
              "      <th>gene_id</th>\n",
              "      <th>gene_name</th>\n",
              "      <th>gene_type</th>\n",
              "      <th>gene_strand</th>\n",
              "      <th>junction_Start</th>\n",
              "      <th>junction_End</th>\n",
              "      <th>output_type</th>\n",
              "      <th>variant_scorer</th>\n",
              "      <th>...</th>\n",
              "      <th>biosample_type</th>\n",
              "      <th>biosample_life_stage</th>\n",
              "      <th>data_source</th>\n",
              "      <th>endedness</th>\n",
              "      <th>genetically_modified</th>\n",
              "      <th>transcription_factor</th>\n",
              "      <th>histone_mark</th>\n",
              "      <th>gtex_tissue</th>\n",
              "      <th>raw_score</th>\n",
              "      <th>quantile_score</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr3:58394738:A&gt;T</td>\n",
              "      <td>chr3:57870450-58919026:.</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>ATAC</td>\n",
              "      <td>CenterMaskScorer(requested_output=ATAC, width=...</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>paired</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>-0.004951</td>\n",
              "      <td>-0.259272</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>168</th>\n",
              "      <td>chr3:58394738:A&gt;T</td>\n",
              "      <td>chr3:57870450-58919026:.</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>DNASE</td>\n",
              "      <td>CenterMaskScorer(requested_output=DNASE, width...</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>paired</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>-0.030459</td>\n",
              "      <td>-0.680793</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2109</th>\n",
              "      <td>chr3:58394738:A&gt;T</td>\n",
              "      <td>chr3:57870450-58919026:.</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>CHIP_HISTONE</td>\n",
              "      <td>CenterMaskScorer(requested_output=CHIP_HISTONE...</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>H3K27ac</td>\n",
              "      <td>NaN</td>\n",
              "      <td>-0.000978</td>\n",
              "      <td>-0.301775</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2110</th>\n",
              "      <td>chr3:58394738:A&gt;T</td>\n",
              "      <td>chr3:57870450-58919026:.</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>CHIP_HISTONE</td>\n",
              "      <td>CenterMaskScorer(requested_output=CHIP_HISTONE...</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>H3K27me3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.000441</td>\n",
              "      <td>0.080577</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2111</th>\n",
              "      <td>chr3:58394738:A&gt;T</td>\n",
              "      <td>chr3:57870450-58919026:.</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>CHIP_HISTONE</td>\n",
              "      <td>CenterMaskScorer(requested_output=CHIP_HISTONE...</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>H3K36me3</td>\n",
              "      <td>NaN</td>\n",
              "      <td>0.000000</td>\n",
              "      <td>-0.092027</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>91363</th>\n",
              "      <td>chr16:1135446:G&gt;T</td>\n",
              "      <td>chr16:611158-1659734:.</td>\n",
              "      <td>ENSG00000290756</td>\n",
              "      <td>ENSG00000290756</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>-</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>RNA_SEQ</td>\n",
              "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td></td>\n",
              "      <td>-0.000391</td>\n",
              "      <td>-0.402670</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>91364</th>\n",
              "      <td>chr16:1135446:G&gt;T</td>\n",
              "      <td>chr16:611158-1659734:.</td>\n",
              "      <td>ENSG00000292400</td>\n",
              "      <td>ENSG00000292400</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>-</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>RNA_SEQ</td>\n",
              "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td></td>\n",
              "      <td>0.000173</td>\n",
              "      <td>0.280661</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>91365</th>\n",
              "      <td>chr16:1135446:G&gt;T</td>\n",
              "      <td>chr16:611158-1659734:.</td>\n",
              "      <td>ENSG00000292423</td>\n",
              "      <td>ENSG00000292423</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>-</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>RNA_SEQ</td>\n",
              "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td></td>\n",
              "      <td>-0.001031</td>\n",
              "      <td>-0.641886</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>91366</th>\n",
              "      <td>chr16:1135446:G&gt;T</td>\n",
              "      <td>chr16:611158-1659734:.</td>\n",
              "      <td>ENSG00000292431</td>\n",
              "      <td>ENSG00000292431</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>-</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>RNA_SEQ</td>\n",
              "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td></td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.103452</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>91367</th>\n",
              "      <td>chr16:1135446:G&gt;T</td>\n",
              "      <td>chr16:611158-1659734:.</td>\n",
              "      <td>ENSG00000292432</td>\n",
              "      <td>ENSG00000292432</td>\n",
              "      <td>lncRNA</td>\n",
              "      <td>-</td>\n",
              "      <td>None</td>\n",
              "      <td>None</td>\n",
              "      <td>RNA_SEQ</td>\n",
              "      <td>GeneMaskLFCScorer(requested_output=RNA_SEQ)</td>\n",
              "      <td>...</td>\n",
              "      <td>primary_cell</td>\n",
              "      <td>adult</td>\n",
              "      <td>encode</td>\n",
              "      <td>single</td>\n",
              "      <td>False</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td></td>\n",
              "      <td>0.000900</td>\n",
              "      <td>0.606708</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>556 rows × 24 columns</p>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-1b643293-2470-4bc3-b01f-3d0ef6f4ecf2')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-1b643293-2470-4bc3-b01f-3d0ef6f4ecf2 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-1b643293-2470-4bc3-b01f-3d0ef6f4ecf2');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "    <div id=\"df-366056fc-c849-409c-8d7d-1682da4a8601\">\n",
              "      <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-366056fc-c849-409c-8d7d-1682da4a8601')\"\n",
              "                title=\"Suggest charts\"\n",
              "                style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "      </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "      <script>\n",
              "        async function quickchart(key) {\n",
              "          const quickchartButtonEl =\n",
              "            document.querySelector('#' + key + ' button');\n",
              "          quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "          quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "          try {\n",
              "            const charts = await google.colab.kernel.invokeFunction(\n",
              "                'suggestCharts', [key], {});\n",
              "          } catch (error) {\n",
              "            console.error('Error during call to suggestCharts:', error);\n",
              "          }\n",
              "          quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "          quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "        }\n",
              "        (() => {\n",
              "          let quickchartButtonEl =\n",
              "            document.querySelector('#df-366056fc-c849-409c-8d7d-1682da4a8601 button');\n",
              "          quickchartButtonEl.style.display =\n",
              "            google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "        })();\n",
              "      </script>\n",
              "    </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe"
            }
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}